File: LC69.cpp
class Solution {
public:
    int mySqrt(int x) {
        int s = 1, e = x/2;

        if(x == 1)
            return x;
        
        while(s <= e){
            int m = s + (e-s)/2;

            if(m == x/m)
                return m;
            
            if(m > x/m)
                e = m - 1;
            else
                s = m + 1;
        }

        return e;
    }
};

File: LC69.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l = 1
        h = x//2
        
        if x == 1:
            return 1
        
        while l <= h:
            m = (l + h)//2
            
            if m == x//m:
                return m
            
            if m > x//m:
                h = m - 1
            else:
                l = m + 1
        
        return h
